### PR TITLE
added methods for creating and executing compound queries with more t…

### PR DESCRIPTION
**Pull request by kevin-bruhwiler**
…han one query

### CODE REVIEW

**Comment by kevin-bruhwiler** [comment]
Can you both take a look at this? There aren't many changes, but I want to make sure that it makes sense to you, and that this framework will work for the new sorts of queries you're putting together.

I can't properly test this until Menuka resolves some inconsistencies with GISJOIN fields in the database (which he's been promising to do since Monday) so I can't verify that it's totally working.

File: src/iframe/js/grpc/GRPC_Querier/grpc_querier.bundle.js [hunk 1]
```diff
@@ -2674,6 +2646,82 @@ SustainQuerier = {
...
+     * @param {Object} secondQuery
+     *        The second query object, if present
+     * @param {Object} secondCompoundRequest
+     *        The second compound request object, if present
+     * @param {Number} join
+     *        The type of join being used (number corresponding the proto definition)
+     * @return {Object}
+     *         The compound request object
+     */
+    makeCompoundQuery: function (firstQuery, firstCompoundRequest, secondQuery, secondCompoundRequest, join=0){
```
Comment (drg101) [inline src/iframe/js/grpc/GRPC_Querier/grpc_querier.bundle.js:2694]: So, in essence, you could create an infinite chain of queries using this function right?

Comment (kevin-bruhwiler) [reply src/iframe/js/grpc/GRPC_Querier/grpc_querier.bundle.js:2694]: That's the idea. Let's you query as many collections as you need to, all at once. It's essentially a binary tree, with each node specifying how to join the leaves below it.

Comment (drg101) [reply src/iframe/js/grpc/GRPC_Querier/grpc_querier.bundle.js:2694]: Seems simple enough, this should make creating new queries with the UI easy too.

**Comment by kevin-bruhwiler** [comment]
Ok, I think this is ready for review now. I've verified that it works and I've added an example query. It's a little cleaner and hopefully easier to write. If you run into any bugs with the new server let me know.

**Review by drg101 — APPROVED** [review]
Looks clean and seems faster than before, good work!

**Pull request merged**